### ISSUE TITLE
Not all devices load when IoT Hub has many devices

### ISSUE THREAD

**Issue by rtheil** [microsoft/vscode-azure-iot-toolkit#537]
Our production IoT Hub has many hundreds of devices in it. In the rare instance when we must push a deployment to a single edge device, we use this plugin to do so. The problem is that the list of devices doesn't contain all the devices in the hub. We use both edge and non-edge devices. Now we've reached a point where many devices do not show up, and we are unable to publish to a single device. Even using command palette we can't just type the name of the device to publish to because it's not in the list.

**Comment by AleksandrLiakhavetsEPAM** [comment]
Did you resolve this issue?

**Comment by rtheil** [comment]
@AleksandrLiakhavetsEPAM No. I've resorted to manually pushing deployments to devices using the azure command prompt.

**Comment by Delapouite** [comment]
Related to this old issue in the previous extension : https://github.com/microsoft/vscode-azure-iot-tools/issues/20

The underlying problem appears to be the limit of 1000 devices.

I agree that in many situation you want to target a specific device and you already know its id.

Therefore, instead of having to browse through a huge list, the UI could offer a way to just type the id of the device and they do whatever you need to perform with it.

**Comment by rtheil** [comment]
I gave up. I use the az command now and have written a powershell script that does all the work for me. This extension is completely useless,

**Comment by rido-min** [comment]
The main goal of this extension is to provide an easy way to interact with hub devices while developing your solution, and it is not intended to be used in production with thousands of devices. 

You should be able to use `az iot` to interact with specific devices, including edge.

**Comment by rtheil** [comment]
It's understandable that it's not a tool meant for production use, but we're beyond that now. We have 1153 devices in our development IoT hub, which was not the case when I originally opened this issue. This tool basically uses the az command to push a deployment to a device. It's just strange to justify the limit based on the idea that it's not meant for production. All this needs is the ability to just enter the deviceId outside of the list.
